api/app/src/mappings/apple/activity.ts

```diff
@@ -0,0 +1,139 @@
+import { Activity } from "@metriport/api";
+import dayjs from "dayjs";
+
+import { AppleHealth, AppleHealthItem, hasActivity } from "../../mappings/apple";
+import { PROVIDER_APPLE } from "../../shared/constants";
+
+export function mapDataToActivity(data: AppleHealth) {
+  const activity: Activity[] = []
+
+  if (hasActivity(data)) {
+    data.HKQuantityTypeIdentifierActiveEnergyBurned?.forEach((appleItem) => {
+      const dateIndex = findDateIndex(activity, appleItem);
+
+      if (dateIndex >= 0) {
+        activity[dateIndex] = {
+          ...activity[dateIndex],
+          summary: {
+            ...activity[dateIndex].summary,
+            energy_expenditure: {
+              ...activity[dateIndex].summary?.energy_expenditure,
+              active_kcal: appleItem.value
+            }
+          }
+        }
+        return;
+      }
+
+      activity.push({
+        metadata: {
+          date: dayjs(appleItem.date).format("YYYY-MM-DD"),
+          source: PROVIDER_APPLE,
+        },
+        summary: {
+          energy_expenditure: {
+            active_kcal: appleItem.value
+          }
+        },
+      })
+    })
+
+    data.HKQuantityTypeIdentifierBasalEnergyBurned?.forEach((appleItem) => {
+      const dateIndex = findDateIndex(activity, appleItem);
+
+      if (dateIndex >= 0) {
+        activity[dateIndex] = {
+          ...activity[dateIndex],
+          summary: {
+            ...activity[dateIndex].summary,
+            energy_expenditure: {
+              ...activity[dateIndex].summary?.energy_expenditure,
+              basal_metabolic_rate_kcal: appleItem.value
+            }
+          }
+        }
+        return;
+      }
+
+      activity.push({
+        metadata: {
+          date: dayjs(appleItem.date).format("YYYY-MM-DD"),
```

Comment:
We have a constant for the literal determining the format of Date ISO: `./shared/date.ts/ISO_DATE`

It would be good to replace this and others.